Include/internal/pycore_tuple.h

```diff
@@ -32,6 +33,39 @@ typedef struct {
     PyTupleObject *it_seq; /* Set to NULL when iterator is exhausted */
 } _PyTupleIterObject;
 
+#define _PyTuple_RESET_HASH_CACHE(op) (_PyTuple_CAST(op)->ob_hash = -1)
```

Comment:
Something like this would be more safer.
```suggestion
#define _PyTuple_RESET_HASH_CACHE(op) \
    do { \
        if (op != NULL) {\
            _PyTuple_CAST(op)->ob_hash = -1); \
         }\
      } while (0)
```